File: app/src/main/java/com/ed/redditapp/ui/activities/subreddit/SubredditDaggerComponent.java
package com.ed.redditapp.ui.activities.subreddit;

import com.ed.redditapp.ActivityScope;
import com.ed.redditapp.AppDaggerComponent;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppDaggerComponent.class,
           modules = SubredditDaggerModule.class)
public interface SubredditDaggerComponent {
    void inject(SubredditActivity activity);
}

File: app/src/main/java/com/ed/redditapp/ui/subreddit/SubredditDaggerComponent.kt
package com.ed.redditapp.ui.subreddit

import com.ed.redditapp.ActivityScope
import com.ed.redditapp.AppDaggerComponent
import dagger.Component

@ActivityScope
@Component(dependencies = [AppDaggerComponent::class],
           modules = [SubredditDaggerModule::class])
interface SubredditDaggerComponent {
    fun inject(activity: SubredditActivity?)
}